sharding/contracts/sharding_manager.sol

```diff
@@ -1,253 +1,326 @@
-    // Check if only one collation in one period
-    assert(periodHead[_shardId] < int(_expectedPeriodNumber));
-
-    // Check the signature with validation_code_addr
-    headerVars.collatorAddr = getEligibleCollator(_shardId, block.number/periodLength);
-    require(headerVars.collatorAddr != 0x0);
-    require(msg.sender == headerVars.collatorAddr);
-
-    // Check score == collationNumber
-    headerVars.score = collationHeaders[_shardId][_parentHash].score + 1;
-    require(_number == headerVars.score);
-
-    // Add the header
-    collationHeaders[_shardId][headerVars.entireHeaderHash] = CollationHeader({
-      parentHash: _parentHash,
-      score: headerVars.score
-    });
+  function register_notary() public payable returns(bool) {
+    address notary_address = msg.sender;
+    require(!notary_registry[notary_address].deposited);
+    require(msg.value == NOTARY_DEPOSIT);
+
+    // update notary_sample_size
+    update_notary_sample_size();
+    
+    uint index;
+    if (!empty_stack()) {
+      index = stack_pop();
+      notary_pool[index] = notary_address;
+    }
+    else {
+      index = notary_pool_len; 
+      notary_pool.push(notary_address);
+    }
+    ++notary_pool_len;
 
-    // Update the latest period number
-    periodHead[_shardId] = int(_expectedPeriodNumber);
+    notary_registry[notary_address] = Notary({
+      deregistered: 0,
+      pool_index: index,
+      deposited: true
+    });
 
-    // Determine the head
-    if (headerVars.score > collationHeaders[_shardId][shardHead[_shardId]].score) {
-      shardHead[_shardId] = headerVars.entireHeaderHash;
-      headerVars.isNewHead = true;
+    // if current index is greater than notary_sample_size, increase notary_sample_size for next period
+    if (index >= next_period_notary_sample_size) {
+      next_period_notary_sample_size = index + 1;
     }
 
-    CollationAdded(_shardId, _expectedPeriodNumber, _periodStartPrevHash,
-                   _parentHash, _transactionRoot, _coinbase, _stateRoot, 
-                   _receiptRoot, _number, headerVars.isNewHead, headerVars.score);
-
+    NotaryRegistered(notary_address, index);
     return true;
   }
 
-  // Records a request to deposit msg.value ETH to address to in shard shard_id
-  // during a future collation. Saves a `receipt ID` for this request,
-  // also saving `msg.sender`, `msg.value`, `to`, `shard_id`, `startgas`,
-  // `gasprice`, and `data`.
-  function txToShard(address _to, int _shardId, uint _txStartgas, uint _txGasprice,
-                     bytes12 _data) public payable returns(int) {
-    receipts[numReceipts] = Receipt({
-      shardId: _shardId,
-      txStartgas: _txStartgas,
-      txGasprice: _txGasprice,
-      value: msg.value,
-      sender: msg.sender,
-      to: _to,
-      data: _data
-    });
-    var receiptId = numReceipts;
-    ++numReceipts;
+  // Removes the notary from the notary pool and sets deregistered period
+  function deregister_notary() public returns(bool) {
```

Comment:
Make these funcs all camelCase plz

Reply:
Functions should use mixedCase.

https://solidity.readthedocs.io/en/v0.3.1/style-guide.html#function-names